autonomic-plugin-common/src/main/java/br/com/autonomiccs/autonomic/plugin/common/guru/AutonomiccsSystemVmsGuru.java

```diff
@@ -197,9 +241,40 @@ public boolean finalizeVirtualMachineProfile(VirtualMachineProfile profile, Depl
         if (dc.getDns2() != null) {
             buf.append(" dns2=").append(dc.getDns2());
         }
-        String bootArgs = buf.toString();
-        logger.debug("Boot Args for " + profile + ": " + bootArgs);
-        return true;
+        return buf.toString();
+    }
+
+    /**
+     * It configure the gateway and network management traffic.
+     */
+    protected void configureNicGatewayAndManagementNetwork(DeployDestination dest, StringBuilder buf, NicProfile nic) {
+        if (nic.isDefaultNic()) {
+            buf.append(" gateway=").append(nic.getIPv4Gateway());
+        }
+
+        if (nic.getTrafficType() == TrafficType.Management) {
+            String mgmt_cidr = configurationDao.getValue(Config.ManagementNetwork.key());
+            if (NetUtils.isValidCIDR(mgmt_cidr)) {
+                buf.append(" mgmtcidr=").append(mgmt_cidr);
+            }
+            buf.append(" localgw=").append(dest.getPod().getGateway());
+        }
+    }
+
+    /**
+     * It configures the IP and mask on a NIC. If the nic has null IP it configures the
```

Comment:
did you mean, if there is no IPV4address set in the VM, it will consider that it is being used an IPV6?